cluster-autoscaler/cloudprovider/gce/gce_reserved.go

```diff
@@ -95,20 +97,23 @@ func (r *GceReserved) CalculateKernelReserved(physicalMemory int64, os Operating
 		if physicalMemory > swiotlbThresholdMemory {
 			reserved += swiotlbReservedMemory
 		}
+		if physicalMemory <= lowMemoryThreshold {
```

Comment:
What's the reasoning for moving this here? Have you verified that it won't mess up cases other than ARM? It seems we could be adding `lowMemoryThreshold` in more cases than before.

Reply:
The new logic is taken as an offset of the "final" previous reserved values. To replicate that the reserved logic, we essentially made it the last code in COS. `lowMemoryThreshold` is moved earlier in code as the rest of the code after this point doesn't rely on `var reserved` (except new logic).

I didn't think this would impact addition cases by adding `lowMemoryThreshold` earlier, as the hierarchy of the if statement is the same as before and the var reserved isn't directly used in any existing code. Also, this impact should have directly affect the unit tests, which are not affected.

Reply:
Oof, I've misread the if condition - I thought it was `reserved <= lowMemoryThreshold`, like a final bump if the `reserved` value was too low, in which case moving it would change things.

Please disregard :stuck_out_tongue: